apps/studio/components/ui/EdgeFunctionBlock/EdgeFunctionBlock.tsx

```diff
@@ -0,0 +1,135 @@
+  /** Toggle visiblity of code on render */
+  showCode?: boolean
+  /** Whether function block is draggable */
+  draggable?: boolean
+  /** Tooltip when hovering over the header of the block */
+  tooltip?: ReactNode
+  /** Optional callback on drag start */
+  onDragStart?: (e: DragEvent<Element>) => void
+}
+
+export const EdgeFunctionBlock = ({
+  label,
+  code,
+  functionName,
+  actions,
+  showCode: _showCode = false,
+  tooltip,
+}: EdgeFunctionBlockProps) => {
+  const { ref } = useParams()
+  const [isDeployed, setIsDeployed] = useState(false)
+
+  const { mutateAsync: deployFunction, isLoading: isDeploying } = useEdgeFunctionDeployMutation({
+    onSuccess: () => {
+      setIsDeployed(true)
+      toast.success('Successfully deployed edge function')
+    },
+  })
+
+  const handleDeploy = async () => {
+    if (!code || isDeploying || !ref) return
+
+    try {
+      const deployResult = await deployFunction({
+        projectRef: ref,
+        metadata: {
+          entrypoint_path: 'index.ts',
+          name: functionName,
+          verify_jwt: false,
+        },
+        files: [{ name: 'index.ts', content: code }],
+      })
+    } catch (error) {
+      toast.error(
+        `Failed to deploy function: ${error instanceof Error ? error.message : 'Unknown error'}`
+      )
+    }
+  }
+
+  const functionUrl = ref ? `https://${ref}.supabase.red/functions/v1/${functionName}` : ''
+
+  return (
+    <ReportBlockContainer
+      tooltip={tooltip}
+      icon={
+        <Code
+          size={18}
+          strokeWidth={1.5}
+          className={cn(
+            'transition-colors fill-foreground-muted group-aria-selected:fill-foreground',
+            'w-5 h-5 shrink-0 grow-0 -ml-0.5'
+          )}
+        />
+      }
+      label={label}
+      actions={
+        <>
+          <Button
+            type="outline"
+            size="tiny"
+            loading={isDeploying}
+            disabled={!ref}
+            onClick={handleDeploy}
+          >
+            {isDeploying ? 'Deploying...' : 'Deploy'}
+          </Button>
+
+          {actions}
+        </>
+      }
+    >
+      <div className="shrink-0 w-full max-h-96 overflow-y-auto">
+        <CodeBlock
+          hideLineNumbers
+          wrapLines={false}
+          value={code}
+          language={'typescript' as CodeBlockLang}
+          className={cn(
+            'max-w-none block !bg-transparent !py-3 !px-3.5 prose dark:prose-dark border-0 text-foreground !rounded-none w-full',
+            '[&>code]:m-0 [&>code>span]:text-foreground'
+          )}
+        />
+      </div>
+
+      {(isDeploying || isDeployed) && (
+        <div className="p-4 w-full border-t bg-surface-75 text-xs">
+          {isDeploying ? (
+            <p className="text-foreground-light">Deploying function...</p>
+          ) : (
+            <>
+              <p className="text-foreground-light mb-2">
+                Function deployed successfully! You can now invoke it at:
```

Comment:
just a feedback - wondering to include CLI commands, so that users can directly copy paste into the terminal to run it - from the POV of someone not familiar with edge functions im not too sure what to do with this URL

![image](https://github.com/user-attachments/assets/1841e390-67e8-4536-b230-d52aedc02ffd)

Reply:
thinking also maybe provide a link to the edge function on the dashboard?